presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -154,6 +155,7 @@ public PlanOptimizers(Metadata metadata, SqlParser sqlParser, FeaturesConfig fea
                                         new PushLimitThroughSemiJoin(),
                                         new MergeLimitWithDistinct(),
 
+                                        new PruneSemiJoinColumns(),
```

Comment:
This is not predicate pushdown migration rule so I think it should not belong to this group.